[inezkok] iP

inezkok opened the pull request:
# Beep Boop Bot

> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes))

Beep Boop Bot frees your mind of having to remember things you need to do. It's
- text-based
- easy to learn
- ~~FAST~~ SUPER FAST to use

All you need to do is
1. download it from [here](https://github.com/inezkok/ip/releases/tag/A-Jar)
2. double-click it
3. add your tasks
4. let it manage your tasks for you :wink:

And it is **FREE**!

Features:
- [x] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

 If you Java programmer, you can use it to practice Java too. Here's the main method:
 ```
 public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

inezkok committed:
Implement Level-0: Rename, Greet, Exit

inezkok committed:
Implement Level-1: Echo

inezkok committed:
Implement Level-2: Add, List

inezkok committed:
Implement Level-3: Mark as Done

inezkok committed:
Implement Level-4: ToDos, Events, Deadlines

inezkok committed:
Implement A-TextUiTesting: Automated Text UI Testing

inezkok committed:
Implement Level-5: Handle Errors

inezkok committed:
Implement Level-6: Delete

inezkok committed:
Implement Level-7: Save

inezkok committed:
Edit exit message

inezkok committed:
Edit greet message

inezkok committed:
Merge branch 'branch-Level-7'

inezkok committed:
Implement Level-8: Dates and Times (without time)

inezkok committed:
Edit DukeException error message

inezkok committed:
Merge branch 'branch-Level-8'

inezkok committed:
Edit exception messages

inezkok committed:
Implement A-MoreOOP

inezkok committed:
Implement A-Packages

inezkok committed:
Merge branch 'add-gradle-support' into branch-A-Gradle

inezkok committed:
Implement A-JUnit

inezkok committed:
Implement A-Jar

inezkok committed:
Implement A-JavaDoc

inezkok committed:
Implement A-CodingStandard

inezkok committed:
Implement Level-9

inezkok committed:
Merge branch 'branch-A-CodingStandard'

inezkok committed:
Merge branch 'branch-Level-9'

inezkok committed:
Correct A-Jar

inezkok committed:
Correct coding standard

inezkok committed:
Implement A-CheckStyle

inezkok committed:
Implement Level-10: GUI

inezkok committed:
Implement GUI with FXML and fix CheckStyle and JavaDoc

inezkok committed:
Fix JavaFX warning

inezkok committed:
Fix text truncating in dialog box

yezkez10 commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,182 @@
+     * @param size The size of the current task list.
+     * @throws DukeException On input error.
+     */
+    private static void validateMarkOrUnmarkorDelete(String number, int size) throws DukeException {
```
Maybe choose a more clear and concise name for this method, ie checkCommandValidity, or checkValidity

yezkez10 commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,182 @@
+     * @param description The string representation of the todo description.
+     * @throws DukeException On input error.
+     */
+    private static void validateToDo(String description) throws DukeException {
```
Maybe use a clearer method name ie checkTodoValidity

yezkez10 commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,95 @@
+        appendOutput(taskString);
+        appendOutput(sizeString);
+    }
+}
```
Great abstraction of message printing!

yezkez10 reviewed (commented):
Looks good to me, some things to note:
Is the abstraction into 10 types of commands the best way? 
Is there a way to make do with lesser classes such that the overall structure is more concise and less confusing for readers?
Also, maybe you could consider changing some of the method names to be instructional so that readers can get a better idea of what each method does at first glance.

CJ-Lee01 commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,182 @@
+     * @throws DukeException On input error.
+     */
+    public static Command parse(String input, int size) throws DukeException {
+        if (input.equals("bye")) {
```
There seems to be a lot of if-else statements here. Would it be better to use a switch statement?

CJ-Lee01 commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,198 @@
+     * @param date The string representation of the date.
+     * @throws DukeException On format error.
+     */
+    private void validateDate(String date) throws DukeException {
```
would it be better if the method returned true or false instead of throwing an exception?

CJ-Lee01 replied:
Actually nvrmind, I can see why you are throwing the duke exception.

CJ-Lee01 commented on src/main/java/duke/command/AddDeadlineCommand.java:
Looks great!
Just a thought: Would be it be good to prevent the same command object from being executed again once it has been executed?

CJ-Lee01 commented on src/main/java/duke/task/Task.java:
I think the tasks are well done!

CJ-Lee01 commented on src/test/java/duke/ParserTest.java:
I like the extensive amount of tests!

However, if you are checking to see if the function does throw an exception, junit has the `assertThrows()` method for checking the type of exceptions being thrown.

inezkok committed:
Update names of method to coding standard

inezkok committed:
Edit application for JAR file

inezkok committed:
Implement A-Assertions

inezkok committed:
Parser class: implement A-CodingQuality

Parser class methods (checkDeadline, checkEvent) have some complicated
expressions, affecting readability.

Let's break down the complicated expressions into steps (create boolean
intermediate steps like isDescriptionBlank).

Breaking the complicated expressions down improve code quality and
readability.

inezkok committed:
Merge pull request #2 from inezkok/branch-A-CodeQuality

Parser class: implement A-CodingQuality

inezkok committed:
Merge branch 'master' into branch-A-Assertions

inezkok committed:
Merge pull request #1 from inezkok/branch-A-Assertions

Branch A-Assertions

inezkok committed:
Find command: make matching case-insensitive

Find command is case-sensitive.

A case-insensitive find is more user-friendly because users cannot be
expected to remember the exact case of the keywords.

Let's change the string representation of each task and keyword to be
in lowercase before checking if they match.

inezkok committed:
Add Increment: A-CI

inezkok committed:
Implement C-Sort

The task list cannot be sorted by their categories.

A sorted task list allows the user to view their tasks in a more
organised manner, making it easier for them to differentiate the tasks
by their categories.

Let's create a TaskCategoryComparator class that implements the
Comparator interface and a SortCategoryCommand class.

Creating a TaskCategoryComparator class allows us to use the sort method
in the Collections class to sort our array list of tasks.

inezkok committed:
Edit import statement

inezkok committed:
Fixed parser checkDeadline and checkEvent methods

inezkok committed:
Reorganise files

inezkok committed:
Improve GUI

inezkok committed:
Improve GUI with error message

inezkok committed:
Fix checkctyle

inezkok committed:
Change colors in GUI

inezkok committed:
Add UI screenshot

inezkok committed:
Add user guide

inezkok committed:
Improve code quality

inezkok committed:
Fix formatDate in Storage and correct user guide

inezkok committed:
Update build.gradle application
